enzy_htp/core/file_system.py

```diff
@@ -166,6 +167,58 @@ def write_data(outfile: str, tag: Any, data: Dict) -> str:
 
     return outfile
 
+def write_data_to_csv(output_directory: str, csv_filename: str = "result.csv", **kwargs):
+    """Write the data to a CSV (.csv) file.
+    
+    Args:
+        output_directory (str): The directory for data output (Absolute path is recommended).
+        csv_filename (str, optional): The filename of the output CSV file.
+        kwargs: The data to be written (keys will be column names).
+    """
+    # Create the output directory if it does not exist
+    safe_mkdir(output_directory)
+    filepath = os.path.join(output_directory, csv_filename)
+
+    # Convert kwargs to DataFrame
+    data_df = pd.DataFrame([kwargs])
+
+    # Append to the CSV file if it exists, otherwise create it with header
+    if not os.path.isfile(filepath):
+        data_df.to_csv(filepath, mode='w', header=True, index=False, encoding='utf-8')
+    else:
+        data_df.to_csv(filepath, mode='a', header=False, index=False, encoding='utf-8')
+
+def write_data_to_excel(output_directory: str, excel_filename: str = "result.xlsx", sheet_name: str = "Sheet1", **kwargs):
+    """Write the data to an excel (.xlsx) file.
+    
+    Args:
+        output_directory (str): The directory for data output (Absolute path is recommended).
+        excel_filename (str, optional): The filename of the output Excel file.
+        sheet_name (str, optional): The sheet name to write data.
+        kwargs: The data to be written (The key strings in kwargs will be column names.).
+    """
+    # Create the directory if it doesn't exist.
+    safe_mkdir(output_directory)
+    filepath = os.path.join(output_directory, excel_filename)
+
+    # Convert kwargs to DataFrame
+    data_df = pd.DataFrame([kwargs])
+    
+    # Check if the file exists
+    if not os.path.isfile(filepath):
+        # Create a new Excel writer object and write to it, because the file does not exist
+        with pd.ExcelWriter(filepath, engine="openpyxl", mode="w") as writer:
+            data_df.to_excel(writer, sheet_name=sheet_name, index=False)
+    else:
+        # Load the existing workbook
+        with pd.ExcelWriter(filepath, engine="openpyxl", mode="a", if_sheet_exists="overlay") as writer:
+            # If the sheet exists, append the data, otherwise write to a new sheet
+            if sheet_name in writer.sheets:
+                startrow = writer.sheets[sheet_name].max_row
+                data_df.to_excel(writer, sheet_name=sheet_name, startrow=startrow, header=False, index=False)
+            else:
+                data_df.to_excel(writer, sheet_name=sheet_name, index=False)
```

Comment:
Append is not a common default behavior. Change the default to overwrite and allow an option to enable append

Reply:
Please resolve this